src/views/pages/Login/index.vue

```diff
@@ -1,30 +1,53 @@
 <template>
   <div class="Login">
-    <h1>{{ msg }}</h1>
+    <h1>Login</h1>
     <div class="login_form">
       <md-field>
         <label>Username</label>
-        <md-input v-model="initial"></md-input>
+        <md-input v-model="credentials.username"></md-input>
       </md-field>
-      <md-field>
+      <md-field :md-toggle-password="false">
         <label>Password</label>
-        <md-input v-model="initial"></md-input>
+        <md-input v-model="credentials.password" type="password"></md-input>
       </md-field>
-      <router-link :to="{ path: '/' }">
-        <md-button class="md-raised md-primary login_button">Login</md-button>
-      </router-link>
+      <div class="login_form_feedback">
+        <p class="error">{{ error }}</p>
+        <md-button v-on:click="login" class="md-raised md-primary login_button">Login</md-button>
+      </div>
     </div>
   </div>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
+const axios = require('axios');
 
 export default Vue.extend({
   name: 'Login',
+  methods: {
+    async login() {
+      console.log('attempting login');
+      const self = this;
```

Comment:
There is no need for this closure here as Vue automatically binds methods to the Vue instance as far as I'm aware